src/main/java/com/hubspot/imap/client/ImapCodec.java

```diff
@@ -38,11 +38,17 @@ public ImapCodec(ImapClientState clientState) {
   @Override
   protected void encode(ChannelHandlerContext ctx, BaseImapCommand msg, List<Object> out) throws Exception {
     String data = msg.commandString();
-    String tag = clientState.getNextTag();
+    if (clientState.getNoTag()) {
```

Comment:
I don't think we should do this based on client state like this (I would try to avoid stateful logic in the client as much as possible). Instead I think we should make it so that the command itself indicates whether or not this tag is necessary.